docker/main.go

```diff
@@ -0,0 +1,199 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/logzio/logzio-go"
+	receiver "github.com/logzio/salesforce-logs-receiver"
+)
+
+const (
+	salesforceURLEnvName     = "SALESFORCE_URL"
+	clientIDEnvName          = "CLIENT_ID"
+	apiVersionEnvName        = "API_VERSION"
+	usernameEnvName          = "USERNAME"
+	passwordEnvName          = "PASSWORD"
+	securityTokenEnvName     = "SECURITY_TOKEN"
+	sObjectTypesEnvName      = "SOBJECT_TYPES"
+	fromTimestampEnvName     = "FROM_TIMESTAMP"
+	intervalEnvName          = "INTERVAL"
+	logzioListenerURLEnvName = "LOGZIO_LISTENER_URL"
+	logzioTokenEnvName       = "LOGZIO_TOKEN"
+
+	defaultLogzioListenerURL = "https://listener.logz.io:8071"
+)
+
+var (
+	errorLogger = log.New(os.Stderr, "ERROR: ", log.Ldate|log.Ltime)
+)
+
+type salesforceCollector struct {
+	receiver *receiver.SalesforceLogsReceiver
+	shipper  *logzio.LogzioSender
+	interval int
+}
+
+func newSalesforceCollector() (*salesforceCollector, error) {
+	rec, err := createSalesforceReceiver()
+	if err != nil {
+		return nil, fmt.Errorf("error creating Salesforce receiver: %w", err)
+	}
+
+	shipper, err := createLogzioSender()
+	if err != nil {
+		return nil, fmt.Errorf("error creating Logz.io sender: %w", err)
+	}
+
+	intervalStr := os.Getenv(intervalEnvName)
+	interval, err := strconv.Atoi(intervalStr)
+	if err != nil {
+		return nil, fmt.Errorf("interval must be a positive number")
+	}
+
+	if interval <= 0 {
+		return nil, fmt.Errorf("interval must be a positive number")
+	}
```

Comment:
Why not creating a default value to use if a user didn't insert a value for it, or if for some reason they inserted a number that's <= 0?